src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -2720,59 +2703,200 @@ public boolean test(Symbol s) {
          }
      }
 
+    /** Report warnings for potentially ambiguous method declarations in the given site.
+     *
+     *  Sometimes ambiguous methods are unaviodable because they're inherited from a
+     *  single supertype. For example, any subtype of Spliterator.OfInt will have
+     *  ambiguities for both forEachRemaining() and tryAdvance() (in both cases the
+     *  overloads are IntConsumer and Consumer&lt;? super Integer&gt;). So we only want
+     *  to "blame" a class when that class is itself responsible for creating the
+     *  ambiguity. So we declare that site is "responsible" for the ambiguity between
+     *  two methods m1 and m2 if there is no direct supertype T of site such that
+     *  m1 and m2, or some overrides thereof, both exist and are ambiguous in T.
+     *  As an optimization, we first check if either method is declared in site and
+     *  does not override any other methods (in which case site is responsible).
+     */
+    void checkPotentiallyAmbiguousOverloads(JCClassDecl tree, Type site) {
+
+        // Skip if warning not enabled
+        if (!lint.isEnabled(LintCategory.OVERLOADS))
+            return;
+
+        // Gather all of site's methods, including overridden methods, grouped by name.
+        // We sort the groups by name simply to eliminate non-determinism in the analysis.
+        List<java.util.List<MethodSymbol>> methodsByName = StreamSupport.stream(
+            types.membersClosure(site, false).getSymbols(new ClashFilter(site), RECURSIVE).spliterator(), false)
+          .map(MethodSymbol.class::cast)
+          .filter(m -> m.owner.type.tsym != syms.objectType.tsym)
```

Comment:
nit: this filter could be folded with the one above, `new ClashFilter(site)`

Reply:
Should be fixed in 7d8a54c6679.